internal/provisioner/plain/controllers/bundleinstance_controller.go

```diff
@@ -262,12 +283,84 @@ func (r *BundleInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		Type:    rukpakv1alpha1.TypeInstalled,
 		Status:  metav1.ConditionTrue,
 		Reason:  rukpakv1alpha1.ReasonInstallationSucceeded,
-		Message: fmt.Sprintf("instantiated bundle %s successfully", bi.Spec.BundleName),
+		Message: fmt.Sprintf("instantiated bundle %s successfully", bundle.GetName()),
 	})
-	bi.Status.InstalledBundleName = bi.Spec.BundleName
+	bi.Status.InstalledBundleName = bundle.GetName()
+
+	if _, err := r.reconcileOldBundles(ctx, oldBundles, bundle); err != nil {
+		meta.SetStatusCondition(&bi.Status.Conditions, metav1.Condition{
+			Type:    rukpakv1alpha1.TypeProgressing,
+			Status:  metav1.ConditionFalse,
+			Reason:  rukpakv1alpha1.ReasonCleanupFailed,
+			Message: err.Error(),
+		})
+		return ctrl.Result{}, err
+	}
+	meta.RemoveStatusCondition(&bi.Status.Conditions, rukpakv1alpha1.TypeProgressing)
```

Comment:
I'm not sure if guidance has changed, but I read that its best to keep _all_ of the possible condition types populated at all times. Essentially, a missing type is interpretted as "Unknown", so if we know that we're not progressing, we should explicitly set it to "False".

Reply:
I ended up dropping bubbling up any status conditions when encountering an error while cleaning up stale Bundles while we figure out the correct UX for this.